google-cloud-firestore/lib/google/cloud/firestore/field_path.rb

```diff
@@ -209,19 +209,15 @@ def hash
 
         protected
 
-        START_FIELD_PATH_CHARS = /\A[a-zA-Z_]/.freeze
         INVALID_FIELD_PATH_CHARS = %r{[~*/\[\]]}.freeze
+        SIMPLE_FIELD_PATH_CHARS = /\A[a-zA-Z_][a-zA-Z_0-9]+\Z/.freeze
 
         def escape_field_for_path field
           field = String field
 
-          if INVALID_FIELD_PATH_CHARS.match(field) ||
-             field["."] || field["`"] || field["\\"]
-            escaped_field = field.gsub(/[`\\]/, "`" => "\\\`", "\\" => "\\\\")
-            return "`#{escaped_field}`"
-          end
+          return field if SIMPLE_FIELD_PATH_CHARS.match field
 
-          return field if START_FIELD_PATH_CHARS.match field
+          field = field.gsub(/[`\\]/, '`' => '\\\`', '\\' => '\\\\')
```

Comment:
I understand that this line was moved from the conditional block above, but I don't think it works correctly:

```
irb(main):024:0> "bak`tik\\".gsub(/[`\\]/, '`' => '\\\`', '\\' => '\\\\')
=> "bak\\\\`tik\\\\"
# Should be: "bak\\`tik\\\\"
```

I believe the replacement value for backticks should be changed to match the nodejs-firestore [implementation](https://github.com/googleapis/nodejs-firestore/blob/master/dev/src/path.ts#L625):

```
field = field.gsub(/[`\\]/, '`' => '\\`', '\\' => '\\\\')
```

Reply:
@quartzmo thanks again!!  You'are right! I just copied in order not to destroy the original Ruby code, and have not noticed that.  I have committed.